### ISSUE TITLE
Question about Request to Lambda function with ARN

### ISSUE THREAD

**Issue by dbbgb** [aws/aws-sdk-cpp#146]
Hi,
I have a question on call lambda functions. In documents says that as the name of the function can be used ARN. 
If I try to use ARN instead of the function name in this example:
```
...
Aws::Lambda::Model::InvokeRequest request;
request.SetInvocationType(InvocationType::RequestResponse);
Aws::Lambda::Model::InvokeResult result;
//request.SetFunctionName("fn_v3"); // It works
request.SetFunctionName("arn:aws:lambda:us-west-2:XXXXXXXXXX:function:fn_v3");  // This does not work
auto payload = Aws::MakeShared<Aws::StringStream>("invokeLambda");
*payload << AWS_TEST_BODY;
request.SetBody(payload);
auto response = lambdaClient.Invoke(request);
...
```
I get the error:
```
Error: Unable to parse ExceptionName: InvalidSignatureException:http://internal.
amazon.com/coral/com.amazon.coral.service/ Message: The request signature we cal
culated does not match the signature you provided. Check your AWS Secret Access
Key and signing method. Consult the service documentation for details.

The Canonical String for this request should have been
'POST /2015-03-31/functions/arn%253Aaws%253Alambda%253Aus-west-2%253AXXXXXXXXX%253A
function%253Afn_v3/invocations
...
```
If I use the function name, the code works.

Help please understand what the problem is.
Thank you.

**Comment by JonathanHenson** [comment]
It looks like we aren't url encoding the ARN correctly and this is causing signature mismatches with AWS Auth. Which revision of the SDK are you using? What is the version number in aws-cpp-sdk-core/include/aws/core/VersionConfig.h   ?

**Comment by rusgl** [comment]
From the configuration file:
#define AWS_SDK_VERSION_STRING "0.9.6-90-g8581d6e"
I updated SDK yesterday from this repository. Now I see that already version 0.10.1 Try it?
Thank you.

**Comment by dbbgb** [comment]
From the configuration file:
#define AWS_SDK_VERSION_STRING "0.9.6-90-g8581d6e"
I updated SDK yesterday from this repository. Now I see that already version 0.10.1 Try it?
Thank you.

**Comment by JonathanHenson** [comment]
Can you turn on logging and grab the path from your HTTP POST to invoke the function so I can track down the encoding bug? Thanks

**Comment by rusgl** [comment]
If I specify only ARN is not working properly the function name and the region (region in ARN -> us-west-2):
Request: https://lambda.us-east-1.amazonaws.com/2015-03-31/functions/arn%3Aaws%3Alambda%3Aus-west-2%3A468131460318%3Afunction%3Amail_send/invocations

If I specify the region 
	Aws::Client::ClientConfiguration config;
	config.region = Aws::Region::US_WEST_2;
	LambdaClient lambdaClient(config);
	
that request as follows:
Request: https://lambda.us-west-2.amazonaws.com/2015-03-31/functions/arn%3Aaws%3Alambda%3Aus-west-2%3A468131460318%3Afunction%3Amail_send/invocations

SDK latest version.
User-Agent: aws-sdk-cpp/0.10.4 Windows/6.1.7601-Service Pack 1 x86

Can I help with something else?
Thank you.

**Comment by dbbgb** [comment]
If I specify only ARN is not working properly the function name and the region (region in ARN -> us-west-2):
Request: https://lambda.us-east-1.amazonaws.com/2015-03-31/functions/arn%3Aaws%3Alambda%3Aus-west-2%3A468131460318%3Afunction%3Amail_send/invocations

If I specify the region:
Aws::Client::ClientConfiguration config;
config.region = Aws::Region::US_WEST_2;
LambdaClient lambdaClient(config);

that request as follows:
Request: https://lambda.us-west-2.amazonaws.com/2015-03-31/functions/arn%3Aaws%3Alambda%3Aus-west-2%3A468131460318%3Afunction%3Amail_send/invocations

SDK latest version.
User-Agent: aws-sdk-cpp/0.10.4 Windows/6.1.7601-Service Pack 1 x86

Can I help with something else?
Thank you.

**Comment by JonathanHenson** [comment]
It looks like the url is getting double encoded from the http layer, I got it to pass by not url escaping the url before making the http call with libcurl. I'm looking for docs that say what curl does under the hood. Apparently, so does windows.

**Comment by JonathanHenson** [comment]
I've fixed this in commit 598d47aaa86eb13a74fe014f367230909afc44ea Please let us know if you need anything else.
